### PR TITLE
Add ADR for bulk data loading

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/TQ6GPOHU/1228-spike-into-loading-governments-from-publishing-api

[Rendered version](https://github.com/alphagov/content-publisher/blob/bulk-data-adr/docs/adr/00XX-bulk-data-loading.md)

### CODE REVIEW

File: docs/adr/00XX-bulk-data-loading.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+These situations where Content Publisher needs all the data records for a
+particular dataset are colloquially referred to as bulk data. This is
+distinguished from other API look ups where we may need to access a
+single resource (such as looking up the topic tags) that have less risks of
+performance penalties.
+
+This particular problem has been faced and solved for GOV.UK before. Whitehall
+takes an [approach](https://github.com/alphagov/whitehall/pull/3298) for the
+topic taxonomy where the data is loaded at runtime from
+[Redis](https://redis.io/). A period background job then refreshes the
```
Comment (leenagupte) [inline docs/adr/00XX-bulk-data-loading.md]: Nitpick: This should be "A periodic..."

File: docs/adr/00XX-bulk-data-loading.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+job to re-populate the Redis data.
+
+This approach makes use of the [Rails Redis cache store][redis-cache-store]
+which is wrapped within an object, [BulkData::Cache][bulk-data-cache], for
+writing and accessing data. We have introduced a concept of
+[repository][government-repository] classes that can be used to work with the
+loading and modelling of data that is stored in the cache.
+
+On a 15 minute interval a [background job][populate-bulk-data-job] will run to
+try re-populate the data. This job will be initially triggered each time the
```
Comment (leenagupte) [inline docs/adr/00XX-bulk-data-loading.md]: Could you try to reword these two sentences as I think they're a little unclear.
For example, what does "job" refer to here? I don't think it's the same as the "background job"?

Comment (kevindew) [reply docs/adr/00XX-bulk-data-loading.md]: Thanks I've rephrased this.

File: docs/adr/00XX-bulk-data-loading.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+job to re-populate the Redis data.
+
+This approach makes use of the [Rails Redis cache store][redis-cache-store]
+which is wrapped within an object, [BulkData::Cache][bulk-data-cache], for
+writing and accessing data. We have introduced a concept of
+[repository][government-repository] classes that can be used to work with the
+loading and modelling of data that is stored in the cache.
+
+On application boot and on a recurring 15 minute interval a
+[background job][populate-bulk-data-job] will run to try re-populate the data.
```
Comment (leenagupte) [inline docs/adr/00XX-bulk-data-loading.md]: Nitpick: Should be "try and" or "try to".

**Pull request merged**